src/TransformerFactory.ts

```diff
@@ -0,0 +1,33 @@
+import * as ts from "typescript";
+
+import {CompilerOptions, LuaTarget} from "./CompilerOptions";
+import {LuaTransformer} from "./LuaTransformer";
+import {LuaTransformer51} from "./targets/LuaTransformer.51";
+import {LuaTransformer52} from "./targets/LuaTransformer.52";
+import {LuaTransformer53} from "./targets/LuaTransformer.53";
+import {LuaTransformerJIT} from "./targets/LuaTransformer.JIT";
+
+export function createTransformer(program: ts.Program): LuaTransformer {
+    const options = program.getCompilerOptions() as CompilerOptions;
+
+    const target = options.luaTarget ? options.luaTarget.toLowerCase() : "";
+
+    let luaTargetTransformer: LuaTransformer;
+
+    switch (target) {
```

Comment:
We were getting rid of version-specific overrides for now right? Seems like only 5.2 has 2 small methods right now